### PR TITLE
ret-3172 implementation

### PR DESCRIPTION
**Pull request by btensay**
https://tools.hmcts.net/jira/browse/RET-3172

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java [hunk 1]
```diff
@@ -282,6 +283,17 @@ public void midEventAmendHearing(CaseData caseData, List<String> errors) {
         }
     }
 
+    public void setScotlandAllocatedOffice(String caseTypeId, CaseData caseData) {
+        if (!SCOTLAND_CASE_TYPE_ID.equals(caseTypeId)) {
+            return;
+        }
+
+        if (caseData.getManagingOffice() == null || UNASSIGNED_OFFICE.equals(caseData.getManagingOffice())) {
+            caseData.setManagingOffice(TribunalOffice.GLASGOW.getOfficeName());
```
Comment (hsjhita) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:292]: not sure on this. If a case comes into Scotland unassigned, this immediately changes the office to Glasgow and removes the logic of having an unassigned office unless I'm reading this wrong

Comment (hsjhita) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:292]: if (caseData.getManagingOffice() == null || !UNASSIGNED_OFFICE.equals(caseData.getManagingOffice())) {
            caseData.setAllocatedOffice(TribunalOffice.GLASGOW.getOfficeName());
        }
        
If the managing office is not unassigned, then assign it to Glasgow by default. If managingOffice = Unassigned, it should not set the allocated office

Comment (btensay) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:292]: I see. So adding allocated office is required only on the assign case event?

Comment (btensay) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:292]: If I have misunderstood the brief, a quick call may be .

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java [hunk 2]
```diff
@@ -282,6 +283,16 @@ public void midEventAmendHearing(CaseData caseData, List<String> errors) {
         }
     }
 
+    public void setScotlandAllocatedOffice(String caseTypeId, CaseData caseData) {
+        if (!SCOTLAND_CASE_TYPE_ID.equals(caseTypeId)) {
+            return;
+        }
+
+        if (caseData.getManagingOffice() == null || !UNASSIGNED_OFFICE.equals(caseData.getManagingOffice())) {
```
Comment (GauravChawlaMOJ) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:291]: should it assign allocated office when managing office is null?

**Pull request closed without merging**